app/policies/user_policy.rb

```diff
@@ -14,4 +14,8 @@ def search?
   def show?
     user&.treasurer? || record == user
   end
+
+  def activities?
+    user&.treasurer? || record == user
```

Comment:
You can only see the activities you ordered something on of your own, or if you are the treasurer you can see it for everyone.